### PR TITLE
Use setTextDocumentLanguage to temporarily set language ID for CUDA sources, and headers

### PR DESCRIPTION
**Pull request by Colengms**
There is a PR on the native side for the bulk of this change.

Adds a new protocol message from the native side which invokes `vscode.languages.setTextDocumentLanguage`.

### CODE REVIEW

**Review by michelleangela — APPROVED** [review]

File: Extension/src/LanguageServer/client.ts [hunk 1]
```diff
@@ -1984,6 +1992,15 @@ export class DefaultClient implements Client {
         }
     }
 
+    private async setTemporaryTextDocumentLanguage(params: SetTemporaryTextDocumentLanguageParams): Promise<void> {
+        // const uri: vscode.Uri = vscode.Uri.file(params.path);
```
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/client.ts]: Why is this here?

**Review by sean-mcmanus — CHANGES_REQUESTED** [review]

**Review by sean-mcmanus — APPROVED** [review]
